### PR TITLE
create gvisor docs

### PR DESCRIPTION
**Pull request by Issif**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind user-interface

/kind content

> /kind translation

> /kind event

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area blog

/area documentation

> /area community

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

#1223

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by leogr** [comment]
/assign @LucaGuerra 

since he's the original author of the gVisor integration in Falco :)

File: content/en/docs/event-sources/gvisor/_index.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+The key observation here is that there is one Sentry process for each gVisor sandbox. If we want them to be able to communicate with Falco, we must set up a form of inter-process communication.
+
+We decided to use a UDS (Unix Domain Socket) to handle the communication between each Sentry and Falco. Falco is acting as a server, and it is responsible for setting up the socket and listening to connections. On the other hand, each Sentry process acts as a client, and it is configured to connect to the endpoint where Falco is listening.
+
+Whenever a syscall gets executed inside the sandboxed application, the Sentry will manage it as usual, plus it will send a message to Falco through the UDS. Messages are serialized through Protocol Buffers so that gVisor and Falco can communicate even if they are written in different programming languages.
+
+Once a message related to a syscall is received, Falco unpacks it and creates the corresponding event in a way that is consumable by our libraries. This way, it is possible to update necessary state information and trigger Falco rules whenever a match occurs!
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:26]: Could you summarize this part? While I agree that it is important to indicate that we connect to every configured gvisor sandboxes via a (configurable) unix domain socket I don't think the documentation page needs to explain the process that brought it to life in detail

File: content/en/docs/event-sources/gvisor/_index.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+![](images/gvisor-02.png)
+
+## Setup gVisor Docker sandbox monitoring with Falco
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:30]: Some of these instructions might be slightly out of date but I can't remember. I know this is a porting of the blog post to make it more maintainable but did you go through the explanation step by step to make sure it works, of course with the latest gVisor and Falco available.

File: content/en/docs/event-sources/gvisor/_index.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+Then, restart the Docker daemon to let it use the new configuration:
+
+```bash
+$ sudo systemctl restart docker
+```
+
+## Runtime detection in action
+
+Now it's time to put everything together and see how to use Falco to monitor gVisor sandboxes.
+To start monitoring gVisor sandboxes, you can use the `-g` or `--gvisor-config` options, passing the path to the pod init config. Falco uses that config file for two main reasons:
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:92]: I believe these options are deprecated and have been replaced by `-o ...` or the Falco configuration file, see the documentation about the modern eBPF: https://falco.org/docs/install-operate/running/#modern-ebpf

File: content/en/docs/event-sources/gvisor/_index.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+- Extract the path of the UDS that needs to be created
+- Create a trace session for all the already existing gVisor sandboxes. New ones will directly connect to the running Falco instance as we configured in the previous step.
+
+### Run Falco stand-alone
+
+Simply run Falco on the command line:
+
+```bash
+$ sudo falco --gvisor-config=/etc/docker/runsc_falco_config.json
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:102]: Same as above

File: content/en/docs/event-sources/gvisor/_index.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+```
+
+You will see Falco alerting:
+```
+07:47:42.173335167: Error File below a known binary directory opened for writing (user=root user_loginuid=0 command=touch touch /bin/foo file=/bin/foo parent=bash pcmdline=bash bash gparent=<NA> container_id=f6d77af4ee3d image=ubuntu) container=f6d77af4ee3d pid=8 tid=8
+```
+
+### Falco and gVisor in action
+
+If you don't happen to have the time to try it right now, here is a short video showing every step to follow.
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:154]: Could you double check that the instructions in this video are still relevant today? If not we should not include it

**Comment by LucaGuerra** [comment]
Thank you so much for this @Issif ! I think it's important to add it to the documentation to make it easier for our adopters to run Falco with gVisor, but I'm a bit afraid that those instructions may need a couple of fixes. If they work as-is, then it's basically all good :)

**Comment by LucaGuerra** [comment]
@Issif did you get a chance to check this out? I can assist with helping set up the environment to run gVisor sandboxes (it's easy to do). It'd be great if you could do it following the instructions so we know if they're accurate as of today

**Comment by Issif** [comment]
> @Issif did you get a chance to check this out? I can assist with helping set up the environment to run gVisor sandboxes (it's easy to do). It'd be great if you could do it following the instructions so we know if they're accurate as of today

I started to work on your comments, summarizing some sections and co. Still haven't found the time to test the steps, I'm preparing a talk for in 2d, I'll see after.

File: content/en/docs/event-sources/gvisor/_index.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+## Run Falco by command line
+
+Simply run Falco by the command line:
+
+```bash
+sudo falco -o "engine.kind=gvisor" -o "engine.gvisor.config=/etc/docker/runsc_falco_config.json"
+```
+
+or edit the file `/etc/falco/falco.yaml` to have these settings and run Falco wiht `sudo falco`:
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:87]:
```suggestion
or edit the file `/etc/falco/falco.yaml` to have these settings and run Falco with `sudo falco`:
```

File: content/en/docs/event-sources/gvisor/_index.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+First, [install **Falco**](https://falco.org/docs/getting-started/installation/) 0.371 or above and [install the **gVisor runsc tool**](https://gvisor.dev/docs/user_guide/install/) 20221122.0 or above:
+
+- Check the Falco version is `>= 0.33.1`:
+  ```bash
+  falco --version
+  ```
+- Check the `runsc` version is `release-20221122.0`:
+  ```bash
+  runsc --version
+  ```
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:35]:
```suggestion
First, [install **Falco**](https://falco.org/docs/getting-started/installation/) and [install the **gVisor runsc tool**](https://gvisor.dev/docs/user_guide/install/). Any version of runsc released in 2023 or later is compatible with Falco.
```

Let's simplify this. Back then we had to specify exact versions because we were using cutting edge builds, now these changes are more stable.

File: content/en/docs/event-sources/gvisor/_index.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+                "--pod-init-config=/etc/docker/runsc_falco_config.json"
+            ]
+        }
+    }
+}
+```
+
+{{% pageinfo color=warning %}}
+Do not forget the comma at the end of the line which sets the `path`.
+{{% /pageinfo %}}
```
Comment (LucaGuerra) [inline content/en/docs/event-sources/gvisor/_index.md:71]: it's json, I don't think we really need this warning.

**Comment by LucaGuerra** [comment]
Thanks a lot @Issif , just a couple minor suggestions

**Review by LucaGuerra — APPROVED** [review]
LGTM
